test/unit/ObjectMakrTests/processFunctions.js

```diff
@@ -0,0 +1,35 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("processes the correct number of functions", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            var inheritance = mocks.mockInheritance();
+
+            // Act
+            objectMaker.functions = {};
```

Comment:
Does it work if you don't reset .functions?

If not, it's better to pass the initial functions to the constructor (in this case via `mocks.mockObjectMakr`).